src/js/controllers/ListCtrl.js

```diff
@@ -277,15 +277,25 @@ function ListCtrl($scope, $route, $routeParams, $location, $http, $filter, authS
   $scope.updateProfile = function (contact, field) {
     var access = field === 'verified' ?  contact.ql.userCanEditVerified : contact.ql.userCanEditKeyContact,
         stateKey = field + 'State';
-    if(!contact._profile.verifiedBy)
-         contact._profile.verified = '';
+
+    if(!contact.verifiedByID){
+       contact.verifiedByID = '';
+       contact.verifiedByName = '';
+       contact.verificationDate = '';
+    }
     if (access) {
       if (contact.ql[stateKey] === 'confirm') {
         contact.userid = contact._profile._userid || contact._profile.userid;
-        contact._profile.verifiedBy = userData.profile.userid;
-
         contact[field] = contact.ql[field];
         contact.ql[stateKey] = 'inProgress';
+        if(contact.verified == true)
```

Comment:
@uiftikhar I do not agree with this: the logic to set the "verifiedBy" variables should be handled at the backend level (ie in hid_profiles). Handling this at the client level and not at least verifying it in the backend would mean that client apps could send out any data (and eventually wrong data) to the backend for storage. The Write logic needs to be handled by hid_profiles, and the client should only be allowed to read this data.